### DISCUSSION TITLE
Feedback on state machine design

### DISCUSSION THREAD

**Discussion by nnduc1994** [statelyai/xstate#4196] [Show and tell]
We have an application that guides user throughout the installation process. The data structure can be sum up below:
  * An installation has multiple question set to be fill in
  * One question set has multiple questions
  * User move from 1 question set to another
  * Questions within questions set can be displayed either by sequence ( one question at the time) or as a list (multiple one at the time) - see the picture attached for better visualize.
  
 I am doing some POC to see if state machine could be utilize here. Here is my draft state machine https://xstate.js.org/viz/?gist=ab8fb80a41df399aa10e15dcf5646fef . What do you guys think about this approach 
 
 **Sequence question**
![Screenshot 2023-08-18 at 12 59 06](https://github.com/statelyai/xstate/assets/7703409/a39450bc-8ecf-4871-92de-45dac7565cdf)

**List Question**
![Screenshot 2023-08-18 at 13 00 53](https://github.com/statelyai/xstate/assets/7703409/693446ba-9e98-4966-beb6-e4f180078e02)

**Comment by davidkpiano** [comment]
I think this is a fine approach. You would use `state.value` to map the state value to the question being shown.
